src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -833,6 +849,46 @@ public static ComponentResultEntity toEntity(LoadFlowResult.ComponentResult comp
                 .bodyToMono(String.class);
     }
 
+    public Mono<String> getSecurityAnalysisStatus(String studyName, String userId) {
+        Objects.requireNonNull(studyName);
+        Objects.requireNonNull(userId);
+
+        return studyRepository.findStudy(userId, studyName).flatMap(entity -> {
+            UUID resultUuid = entity.getSecurityAnalysisResultUuid();
+            return Mono.justOrEmpty(resultUuid).flatMap(uuid -> {
```

Comment:
when can resultUuid be empty ? warning: flatmap doesn't execute in this case

Reply:
Shouldn't it error when the uuid is null, instead of returning an empty mono ?

Reply:
Agree, if no security analysis result in the DB, it means the API is not used correctly and we should throw an error.

Reply:
Here, it's the same thing as getting the security analysis result : it can be called on study creation, when no security analysis has been launched ...

Reply:
Ok